src/modules/cmdNotFound/CmdNotFoundModuleInterface/WinGetCommandNotFound.cs

```diff
@@ -0,0 +1,303 @@
+        {
+            object instance = null;
+            if (IsAdministrator())
+            {
+                var hr = WinGetServerManualActivation_CreateInstance(type.GUID, iid, 0, out instance);
+                if (hr < 0)
+                {
+                    throw new COMException($"Failed to create instance: {hr}", hr);
+                }
+            }
+            else
+            {
+                instance = Activator.CreateInstance(type);
+            }
+
+            IntPtr pointer = Marshal.GetIUnknownForObject(instance);
+            return WinRT.MarshalInterface<T>.FromAbi(pointer);
+        }
+
+        [DllImport("winrtact.dll", EntryPoint = "WinGetServerManualActivation_CreateInstance", ExactSpelling = true, PreserveSig = true)]
+        private static extern int WinGetServerManualActivation_CreateInstance(
+                [In, MarshalAs(UnmanagedType.LPStruct)] Guid clsid,
+                [In, MarshalAs(UnmanagedType.LPStruct)] Guid iid,
+                uint flags,
+                [Out, MarshalAs(UnmanagedType.IUnknown)] out object instance);
+
+        [DllImport("winrtact.dll", EntryPoint = "winrtact_Initialize", ExactSpelling = true, PreserveSig = true)]
+        public static extern void InitializeUndockedRegFreeWinRT();
+
+        public static PackageManager CreatePackageManager()
+        {
+            return Create<PackageManager>(PackageManagerType, PackageManagerIid);
+        }
+
+        public static FindPackagesOptions CreateFindPackagesOptions()
+        {
+            return Create<FindPackagesOptions>(FindPackagesOptionsType, FindPackagesOptionsIid);
+        }
+
+        public static PackageMatchFilter CreatePackageMatchFilter()
+        {
+            return Create<PackageMatchFilter>(PackageMatchFilterType, PackageMatchFilterIid);
+        }
+    }
+
+    public sealed class WinGetComObjects
+    {
+        public static WinGetComObjects Singleton { get; } = new WinGetComObjects();
+
+        private WinGetComObjects()
+        {
+            ComObjectFactory.InitializeUndockedRegFreeWinRT();
+            packageManager = ComObjectFactory.CreatePackageManager();
+            findPackagesOptions = ComObjectFactory.CreateFindPackagesOptions();
+            packageMatchFilter = ComObjectFactory.CreatePackageMatchFilter();
+        }
+
+        public PackageManager packageManager { get; }
+        public FindPackagesOptions findPackagesOptions { get; }
+        public PackageMatchFilter packageMatchFilter { get; }
+    }
+
+    public sealed class Init : IModuleAssemblyInitializer, IModuleAssemblyCleanup
+    {
+        internal const string id = "e5351aa4-dfde-4d4d-bf0f-1a2f5a37d8d6";
+
+        public void OnImport()
+        {
+            if (!Platform.IsWindows)
+            {
+                return;
+            }
+
+            // Ensure WinGet is installed
+            using (var rs = RunspaceFactory.CreateRunspace(InitialSessionState.CreateDefault()))
```

Comment:
`OnImport` always happen on the PS pipeline thread, so you can do the following instead, which is way more efficient than opening a new `Runspace`.
```
using (var pwsh = PowerShell.Create(RunspaceMode.CurrentRunspace))
{
    var results = pwsh.AddCommand("Get-Command")
        .AddParameter("Name", "winget")
        .AddParameter("CommandType", "Application")
        .Invoke();

    if (restuls.Count is 0)
    {
        return;
    }
}
```